packages/gensx-storage/src/database/context.ts

```diff
@@ -0,0 +1,40 @@
+import { createContext, useContext } from "@gensx/core";
+
+import { Database, DatabaseStorage } from "./types.js";
+
+/**
+ * Create the database storage context
+ */
+export const DatabaseContext = createContext<DatabaseStorage | null>(null);
+
+/**
+ * Hook to access the database storage instance
+ * @returns The database storage instance
+ * @throws Error if used outside of a DatabaseProvider
+ */
+export function useDatabaseStorage(): DatabaseStorage {
+  const storage = useContext(DatabaseContext);
+
+  if (!storage) {
+    throw new Error(
+      "useDatabaseStorage must be used within a DatabaseProvider. Wrap your component tree with a DatabaseProvider.",
+    );
+  }
+
+  return storage;
+}
+
+/**
+ * Hook to access a database
+ * @param name The name of the database to access
+ * @returns A promise that resolves to a database object for the given name
+ * @throws Error if used outside of a DatabaseProvider
+ */
+export async function useDatabase(name: string): Promise<Database> {
+  const storage = useDatabaseStorage();
+
+  // Ensure the database exists before returning it
+  await storage.ensureDatabase(name);
```

Comment:
`useDatabase` will be called a lot right? Basically anytime they use the database inside a component? It feels like this might add a lot of latency.

We can probably cache this, or something, so it just does it once, the first time the database is used inside the workflow.

Reply:
oh yeah that's a good point. Curious your thoughts on how we should approach this @EvanBoyle

Reply:
Yes, it will be called whenever a component does 'useDatabase`. 

We could maintain a map of databases that have already been `used` in memory. But cache invalidation is always hard - remembering to delete from the cache on delete, etc. This is why I avoided storing clients in the API. We can add it now, or always do it later. I'm not strongly opinionated.

Reply:
went ahead and added support for this. We were already keeping a list of databases in memory so i just added a method to check if it's ensured and then i make sure we add it to the list when it's ensured and remove it when it's deleted.